lib/Basics/NumberOfCores.cpp

```diff
@@ -45,12 +47,29 @@ std::size_t numberOfCoresImpl() {
     n = 0;
   }
 
-  if (n > 0) {
-    return static_cast<std::size_t>(n);
+  if (n <= 0)
+#endif
+    {
+      n = std::thread::hardware_concurrency();
+    }
```

Comment:
I don't think this code is valid in case `TRI_SC_NPROCESSORS_ONLN` is not defined. 
The code then will be just
``` 
std::size_t numberOfCoresImpl() {
    {
      n = std::thread::hardware_concurrency();
    }
...
``` 
so the compiler should complain about an undefined variable `n`.